### ISSUE TITLE
Flickr engine is broken

### ISSUE THREAD

**Issue by dalf** [searx/searx#315]
The html seems to have changed, but it's seems there is [REST API](https://api.flickr.com/services/rest?sort=relevance&parse_tags=1&content_type=7&extras=can_comment%2Ccount_comments%2Ccount_faves%2Cisfavorite%2Clicense%2Cmedia%2Cneeds_interstitial%2Cowner_name%2Cpath_alias%2Crealname%2Crotation%2Curl_c%2Curl_l%2Curl_m%2Curl_n%2Curl_q%2Curl_s%2Curl_sq%2Curl_t%2Curl_z&per_page=25&page=1&lang=en-US&rb=1&text=proxy&viewerNSID=&method=flickr.photos.search&csrf=&api_key=3e5918155f464baad83cce2efcf8b57e&format=json&hermes=1&hermesClient=1&reqId=rgb38n1&nojsoncallback=1)

In all parameters there is an api_key : I don't know how long it is valid, in which condition.
The call to this URL is trigger inside another minified javascript.

**Comment by Cqoicebordel** [comment]
Yes, normally, the flickr_engine.py use the REST API, with a key you had to get from Flickr.
But searx.me uses the flicknoapi_engine.py, which uses the HTML part of the site.

So, unless you want to get an API KEY from Flickr, we have to manage all with the changes on the HTML part.

**Comment by dalf** [comment]
What I mean: the flicknoapi_engine.py "browses" the HTML page intended to the browser. Now this HTML page uses a REST API with api key, and doesn't contains reference to the thumbnail etc... (not sure, I made a quick look).

Display a [this search](https://www.flickr.com/search/?q=proxy) with chrome dev tools / firebug, you will see a request to /rest..... with the results.

**Comment by Cqoicebordel** [comment]
Ah yes, I understand what you mean.

But on my side, there is no request to the rest api.  The page still contains the api response directly in it, like it was when I wrote flickrnoapi_engine. Sure the regex may need to be changed (I don't have the time to look deeply into it right now), but globally, not much have changed.

I confirmed it with Chromium, Firefox and Opera 12.

So maybe it come from the user agent, or from the beta search in Flickr. But if you do a wget on the page, you'll get the "blob" with all the stuff that the noapi engine parse. You'll see it around line 2644.

EDIT : Oh and I forgot, try to load the page without JS, it might help.
